subxt/src/backend/mod.rs

```diff
@@ -734,4 +857,600 @@ mod test {
+                        Ok(operation_continue("Id1")),
+                    ])),
+                ),
+                (
+                    "chainHead_v1_storage",
+                    Message::Many(Ok(vec![
+                        Ok(storage_items("Id1", &[storage_result("ID2", "Data2")])),
+                        Ok(operation_continue("Id1")),
+                    ])),
+                ),
+                (
+                    "chainHead_v1_storage",
+                    Message::Many(Ok(vec![
+                        Ok(storage_items("Id1", &[storage_result("ID3", "Data3")])),
+                        Ok(storage_done("Id1")),
+                    ])),
+                ),
+            ];
+            let rpc_client = setup_mock_rpc_client(false)
+                .add_method(
+                    "chainHead_v1_storage",
+                    Box::new(|data, sub, _| {
+                        Box::pin(async move {
+                            let response = data.pop("method_response".into()).unwrap_single();
+                            if response.is_ok() {
+                                let item = data.pop("chainHead_v1_storage".into());
+                                if let Some(sub) = sub {
+                                    let item = item;
+                                    sub.write_delayed(item).await
+                                }
+                            }
+                            response.map(|x| RawValue::from_string(x).unwrap())
+                        })
+                    }),
+                )
+                .add_method(
+                    "chainHead_v1_continue",
+                    Box::new(|data, sub, _| {
+                        Box::pin(async move {
+                            let response = data.pop("continue_response".into()).unwrap_single();
+                            if response.is_ok() {
+                                let item = data.pop("chainHead_v1_storage".into());
+                                if let Some(sub) = sub {
+                                    let item = item;
+                                    sub.write_delayed(item).await
+                                }
+                            }
+                            response.map(|x| RawValue::from_string(x).unwrap())
+                        })
+                    }),
+                )
+                .add_mock_data(mock_data)
+                .add_mock_data(response_data)
+                .add_mock_data(continue_data)
+                .build();
+            let backend = run_backend_spawn_background(rpc_client);
+
+            // We try again and should fail mid way
+            let response = backend
+                .storage_fetch_values(
+                    ["ID1".into(), "ID2".into(), "ID3".into()].into(),
+                    crate::utils::H256::random(),
+                )
+                .await
+                .unwrap();
+            // operation returned FollowEvent::OperationError
```

Comment:
dq: Does subxt retry on OperationError?

Reply:
No, all errors are except DisconnectWillReconnect and Rejected are handed back to the user to deal with.